libs/utils/wa_results_collector.py

```diff
@@ -146,9 +146,11 @@ def __init__(self, base_dir=None, wa_dirs=".*", platform=None,
         self.use_cached_trace_metrics = use_cached_trace_metrics
 
         df = pd.DataFrame()
+        df_list = []
         for wa_dir in wa_dirs:
             self._log.info("Reading wa_dir %s", wa_dir)
-            df = df.append(self._read_wa_dir(wa_dir))
+            df_list.append(self._read_wa_dir(wa_dir))
+        df = df.append(df_list)
```

Comment:
If by any chance we end up with an empty df_list... we should add a check here, i.e.:
```python
if len(df_list):
    df = df.append(df_list)
```